rules/prefer-flat.js

```diff
@@ -36,10 +36,15 @@ module.exports = {
                     }
                 });
             },
-            'CallExpression[callee.type="MemberExpression"][callee.property.name="reduce"] > *:function > CallExpression[callee.type="MemberExpression"][callee.property.name="concat"]'(node) {
-                if(node.parent.parent.arguments.length > SECOND && node.parent.parent.arguments[SECOND].type === "ArrayExpression" &&
-                    firstElement(node.arguments).name === node.parent.params[SECOND].name &&
-                    node.callee.object.name === firstElement(node.parent.params).name) {
+            'CallExpression[callee.type="MemberExpression"][callee.property.name="reduce"][arguments.length=2][arguments.1.type=ArrayExpression][arguments.1.elements.length=0] > *:function[params.length=2][params.0.type=Identifier][params.1.type=Identifier] > CallExpression[callee.type="MemberExpression"][callee.property.name="concat"][arguments.length=1][arguments.0.type=Identifier]'(node) {
```

Comment:
Oh wow, I didn't expect this much to be possible within the selector. Nice work!